### PR TITLE
Fix ImageHDU.scale with float

### LINKED ISSUE #6845: Converting float64 to float32 rounds off data in hdu object
Hi,
Why do we round off the data to integers before changing the data type in this line?
https://github.com/astropy/astropy/blob/5f88a0ede0d03cf6329d9d5e0971d540275a8b55/astropy/io/fits/hdu/image.py#L549

To save hardisk space, I wanted to convert my float64 fits images to float32.
I used the `hdulist[0].scale('float32')` method to do that.
However, instead of changing float64 to float32, it rounded off to nearest integer.

### PR DESCRIPTION
**Pull request by saimn**
Fix #6845

### Description

This pull request is to address ...

Fixes #<Issue Number>

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by pllim** [comment]
Running "exotic archs" too, just in case. 🤞

**Review by pllim — APPROVED** [review]
Thanks!

File: astropy/io/fits/hdu/image.py [hunk 1]
```diff
@@ -577,7 +577,10 @@ def _scale_internal(
             self._header["BLANK"] = blank
 
         if self.data.dtype.type != _type:
-            self.data = np.array(np.around(self.data), dtype=_type)
+            if issubclass(_type, np.floating):
+                self.data = np.array(self.data, dtype=_type)
+            else:
+                self.data = np.array(np.around(self.data), dtype=_type)
```
Comment (neutrinoceros) [inline astropy/io/fits/hdu/image.py:583]: I couldn't help but observe that this does a potentially wasteful copy, which could be avoided as
```python
self.data = np.around(self.data).astype(_type, copy=COPY_IF_NEEDED)
```

Comment (neutrinoceros) [reply astropy/io/fits/hdu/image.py:583]: Oh I see. I missed that copying wasn't avoidable in this case.

**Pull request merged**